include/jwt/jwt.hpp

```diff
@@ -72,7 +72,6 @@ inline enum type str_to_type(const jwt::string_view typ) noexcept
   else if(!strcasecmp(typ.data(), "none")) return type::NONE;
 
   JWT_NOT_REACHED("Code not reached");
-  return type::NONE;
```

Comment:
I don't think we should be removing this assuming that the assertion will hit.  There are people out there who disable assertions. Actually it is wierd that it builds with this change.

Reply:
The `JWT_NOT_REACHED` macro always expands to forced UB (__builtin_unreachable or __assume(0)). Because of the UB the compiler is free to assume that no code after the macro will be reached even if we put the `return type::NONE;` here. That is also why it compiles with a warning "unreachable statement".

Reply:
Or in other words, putting the return statement here has no effect. Because of the UB the compiler will just optimize it away.

Reply:
Interesting. Does it still give the assertion with NDEBUG ? Maybe thats a g++ only thing, not sure.

Reply:
The `unreachable` is always active regardless of NDEBUG: https://github.com/arun11299/cpp-jwt/blob/master/include/jwt/assertions.hpp
NDEBUG only adds an `assert` before it.

Reply:
Also regardless of the compiler, because for compilers that are not covered by assertions.hpp the program will not compile, due to the missing definition of the `JWT_NOT_REACHED_MARKER` macro